src/core/src/lib.rs

```diff
@@ -238,3 +239,107 @@ pub trait Device: Sized {
     /// Cleanup unused resources. This should be called between frames. 
     fn cleanup(&mut self);
 }
+
+
+/// An `Instance` holds per-application state for a specific backend
+pub trait Instance {
+    /// Associated `Adapter` type.
+    type Adapter: Adapter;
```

Comment:
There is a single adapter type associated, which is more like a backend abstraction for all adapters.
Adapters then define hardware devices and virtual devices and there capabilities (supported queue types).
I don't really see the issue here unfortunately.